src/job-execution/WebJobExecutor.ts

```diff
@@ -102,29 +103,30 @@ export class WebJobExecutor extends BaseJobExecutor {
 
     const requestPromise = new Promise((resolve, reject) => {
       this.requestClient!.post(apiUrl, formData, undefined)
-        .then(async (res) => {
+        .then(async (res: any) => {
           if (this.serverType === ServerType.SasViya && config.debug) {
             const jsonResponse = await parseSasViyaDebugResponse(
-              res.result as string,
+              res.result,
               this.requestClient,
               this.serverUrl
             )
             this.appendRequest(res, sasJob, config.debug)
             resolve(jsonResponse)
           }
           if (this.serverType === ServerType.Sas9 && config.debug) {
-            const jsonResponse = parseWeboutResponse(res.result as string)
-            if (jsonResponse === '') {
-              throw new Error(
-                'Valid JSON could not be extracted from response.'
-              )
+            let jsonResponse
+            if (typeof res.result === 'string') {
+              jsonResponse = parseWeboutResponse(res.result)
+              if (jsonResponse === '') throw new WeboutResponseError(apiUrl)
```

Comment:
I think that it makes more sense to throw error from `parseWeboutResponse`

Reply:
we wouldn't have access to apiUrl in parseWeboutResponse function.